core/src/main/scala/filodb.core/binaryrecord2/RecordSchema.scala

```diff
@@ -419,22 +419,35 @@ object RecordSchema {
                                                        MapColumn -> 4,
                                                        HistogramColumn -> 4)
 
+  private val tlKeyBuf = new ThreadLocal[Array[Byte]]()
+  private[binaryrecord2] def keyBuf: Array[Byte] = tlKeyBuf.get match {
+    case UnsafeUtils.ZeroPointer => val buf = new Array[Byte](8)
+                                    tlKeyBuf.set(buf)
+                                    buf
+    case b: Array[Byte]          => b
+  }
+
+  // Creates a Long by copying no more than 8 bytes from the bytes array, and zeroing out the rest
+  private def eightBytesToLong(bytes: Array[Byte], index: Int, len: Int): Long = {
+    val buf = keyBuf
```

Comment:
I don't think this approach is any faster than a simpler approach (at least probably not that much). Read one byte at a time from the array and shift+xor it into a long. XXHash implementations already support shorter lengths, I thought.

Reply:
That's true, let me replace this with the shift-XOR.  No state and probably just as fast.

XXHash does support shorter lengths, but the thought is that an exact match is probably just as fast and better than a hash.  Can discuss in person if you want.